src/lib/user_command_input/user_command_input.ml

```diff
@@ -0,0 +1,123 @@
+open Coda_base
+open Signature_lib
+open Coda_numbers
+open Core_kernel
+open Async_kernel
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    type t =
+      { sender: Public_key.Compressed.Stable.V1.t
```

Comment:
There's a dependency cycle between `secrets` and `coda_base`. I could expose `Poly.t` from `user_command.ml` to define the types in this file

Reply:
I'll try to move around the modules to see if I can do that

Reply:
>  I could expose Poly.t from user_command.ml to define the types in this file

Okay. The relevant `Poly.t`s are already exposed, so doing it here is straighforward:

```ocaml
module Payload = struct
  open User_command

  module Common = struct
    [%%versioned
    module Stable = struct
    type t =
      ( Currency.Fee.Stable.V1.t
      , Coda_numbers.Account_nonce.Stable.V1.t option
      , Coda_numbers.Global_slot.Stable.V1.t
      , User_command_memo.Stable.V1.t )
      Payload.Common.Poly.Stable.V1.t
    end]

     type t =
       ( Currency.Fee.t
       , Coda_numbers.Account_nonce.t option
       , Coda_numbers.Global_slot.t
       , Coda_numbers.Global_slot.t
       , User_command_memo.t )
       Payload.Common.Poly.t
   end
  
  [%%versioned
  module Stable = struct
    module V1 = struct
      type t =
        ( Common.Stable.V1.t
        , Payload.Body.Stable.V1.t )
        Payload.Poly.Stable.V1.t
     end
   end]
  type t =
    (Common.t, Payload.Body.t) Payload.Poly.t
end

[%%versioned
module Stable = struct
  module V1 = struct
    type t =
      ( Payload.Stable.V1.t
      , Public_key.Compressed.Stable.V1.t
      , [ `Signature of Signature.Stable.V1.t
        | `Hd_index of Unsigned_extended.UInt32.Stable.V1.t
        | `Keypair of Keypair.Stable.V1.t ] )
      User_Command.Stable.V1.t
  end
end]

type t =
  ( Payload.t
  , Public_key.Compressed.t
  , [ `Signature of Signature.t
    | `Hd_index of Unsigned_extended.UInt32.t
    | `Keypair of Keypair.t ] )
  User_Command.t
```

Reply:
It would also be good to split the conversion across these modules: converting the payload separately (`Payload.to_user_command_payload`) from signing would make the code below much clearer.

Reply:
Yes, already defined the types that way. Refactored the `to_user_command` function